### PR TITLE
Regular updates (Apr 6)

### PR DESCRIPTION
**Pull request by jorgeorpinel**
- Address https://github.com/iterative/dvc/pull/3566#pullrequestreview-388346381
- Address https://github.com/iterative/dvc.org/pull/1096#pullrequestreview-387865301 about `remote add -d`
- Review external data user guide and shared dev server use case to clarify shared/external cache concept
- Complement https://github.com/iterative/dvc.org/pull/1090#pullrequestreview-386203081 (import/update --rev changes)
- Misc. stuff

### CODE REVIEW

File: content/docs/use-cases/shared-development-server.md [hunk 1]
```diff
@@ -3,102 +3,110 @@
...
+so that they can be accessed by your colleagues (assuming their users are
+members of the same group):
 
 ```dvc
-$ sudo find /path/to/dvc-cache -type f -exec chmod 0664 {} \;
-$ sudo find /path/to/dvc-cache -type d -exec chmod 0775 {} \;
-$ sudo chown -R myuser:ourgroup /path/to/dvc-cache/
+$ sudo find /home/shared/dvc-cache -type d -exec chmod 0775 {} \;
+$ sudo find /home/shared/dvc-cache -type f -exec chmod 0664 {} \;
+$ sudo chown -R myuser:ourgroup /home/shared/dvc-cache/
```
Comment (jorgeorpinel) [inline content/docs/use-cases/shared-development-server.md:49]: I'm not sure these commands will always work. What if the shared cache dir has some files your user can't access (if someone didn't do this process before, for example)? Wouldn't the command fail?

File: content/docs/use-cases/shared-development-server.md [hunk 2]
```diff
@@ -3,102 +3,110 @@
...
 
-Tell DVC to use the directory we've set up above as an shared cache location by
-running:
+Tell DVC to use the directory we've set up above as the <abbr>cache</abbr> for
+your <abbr>project</abbr>:
 
 ```dvc
-$ dvc config cache.dir /path/to/dvc-cache
+$ dvc config cache.dir /home/shared/dvc-cache
 ```
```
Comment (jorgeorpinel) [inline content/docs/use-cases/shared-development-server.md:59]: Why do we have all these exact and reproducible command guides in a use case? This section seems more like a user guide to me.

File: content/docs/user-guide/managing-external-data.md [hunk 3]
```diff
@@ -29,22 +29,22 @@ supported:
...
 
-> Avoid using the same remote location that you are using for `dvc push`,
-> `dvc pull`, `dvc fetch` as external cache for your external outputs, because
-> it may cause possible file hash overlaps: The hash value of a data file in
-> external storage could collide with that generated locally for another file.
+> Avoid using the same location of the
+> [remote storage](/doc/command-reference/remote) that you have for `dvc push`
+> and `dvc pull` for external outputs or as external cache, because it may cause
+> file hash overlaps: The hash value of a data file in external storage could
+> collide with the one generated locally for another file.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/managing-external-data.md:41]: Improved this note but it's still pretty strange. Difficult to understand what it is about. The ambiguity of the term "remote" is giving us problems here. Not sure how to address, maybe just remove the note?

File: content/docs/command-reference/import.md [hunk 4]
```diff
@@ -84,9 +84,9 @@ data artifact from the source repo.
   download the file or directory from. The latest commit in `master` (tip of the
   default branch) is used by default when this option is not specified.
 
-  > Note that this adds a `rev` field in the import stage that fixes it to this
-  > revision. This can impact the behavior of `dvc update`. (See
-  > **re-importing** example below.)
+  > Note that this adds a `rev` field in the import stage that fixes it to the
+  > revision. This can impact the behavior of `dvc update`. (See the **Importing
+  > and updating fixed revisions** example below.)
```
Comment (shcheklein) [inline content/docs/command-reference/import.md:89]: minor - make it a link?

File: content/docs/command-reference/update.md [hunk 5]
```diff
@@ -82,15 +81,19 @@ stable.
 > [DVC-file](/doc/user-guide/dvc-file-format) (when there are changes to bring
 > in).
 
-## Example: Updating imported artifacts to a specified revision
+## Example: Updating fixed-revision imports to a different version
```
Comment (shcheklein) [inline content/docs/command-reference/update.md:84]: `fixed-revision imports` sounds very technical :)

File: content/docs/use-cases/shared-development-server.md [hunk 6]
```diff
@@ -3,102 +3,110 @@
...
-$ mkdir -p /path/to/dvc-cache
+$ mkdir -p /home/shared/dvc-cache
 ```
 
-You will have to make sure that the directory has proper permissions setup, so
-that all your colleagues can read and write to it, and can access cache files
-written by others. The most straightforward way to do this is to make sure that
-everyone's users are members of the same group, and that your shared cache
-directory is owned by this group, with the aforementioned permissions.
+> The `/home/shared` directory used as example above is typical in Linux
```
Comment (shcheklein) [inline content/docs/use-cases/shared-development-server.md:21]: no, let's not include notes like this :)

File: content/docs/use-cases/shared-development-server.md (hunk 2 above)
Comment (shcheklein) [reply content/docs/use-cases/shared-development-server.md:59]: yes, I agree ... most of it should be part of the `shared cache` article in the UG. And use case should be probably more generic "Optimizing data management" which can cover this one (multiple people working on a single box), k8s scenario - multiple machines + NAS, etc. We are long overdue on this. There were contributors that made an attempt, but didn't get it right.

File: content/docs/user-guide/managing-external-data.md (hunk 3 above)
Comment (shcheklein) [reply content/docs/user-guide/managing-external-data.md:41]: yep, it's a very complicated to simplify it :) the whole topic is very advanced and very technical. Let's keep it for now with your revisions and get back to it when we have more time to spend on all this data management problems.

**Comment by shcheklein** [comment]
@jorgeorpinel just minor comments, overall looks good to merge.

File: content/docs/command-reference/update.md (hunk 5 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/update.md:84]: Changing to "Updating fixed revisions to a different version".

File: content/docs/use-cases/shared-development-server.md (hunk 6 above)
Comment (jorgeorpinel) [reply content/docs/use-cases/shared-development-server.md:21]: Ok, reverting. But I just copied it from https://dvc.org/doc/user-guide/managing-external-data#examples where it's already being used. Should I remove that one too?

Comment (jorgeorpinel) [reply content/docs/use-cases/shared-development-server.md]: I'll just remove it from there too.

Comment (shcheklein) [reply content/docs/use-cases/shared-development-server.md]: 👍

File: content/docs/use-cases/shared-development-server.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/use-cases/shared-development-server.md:56]: OK. I see we have #429 for something like this already. And #986 is also related.  Should we try to consolidate them into a single issue and include the comments from this review?

Comment (shcheklein) [reply content/docs/use-cases/shared-development-server.md:56]: so, these two mostly about improving the existing case - but they are relevant, up to you to consolidate or not ... there should be tickets related to this from other angles - shared cache on NAS/NFS for example

Comment (shcheklein) [reply content/docs/use-cases/shared-development-server.md:56]: Some other related links:

https://discuss.dvc.org/t/setup-dvc-to-work-with-shared-data-on-nas-server/180/4
https://github.com/iterative/dvc.org/pull/732
https://github.com/iterative/dvc.org/pull/455
https://github.com/iterative/dvc.org/issues/563

and others ... someone should a first step systematize and suggest the way to structure this properly

**Pull request merged**